File: jvm/src/main/kotlin/com/jetbrains/signatureverifier/serialization/RdNInfo.kt
package com.jetbrains.signatureverifier.serialization

import kotlinx.serialization.Serializable
import org.bouncycastle.asn1.ASN1Primitive

@Serializable
data class rdNInfo(
  val type: TextualInfo,
  val value: TextualInfo
) : EncodableInfo {
  override fun toPrimitive(): ASN1Primitive =
    listOf(
      type.toPrimitive(),
      value.toPrimitive()
    ).toDLSequence().toASN1Primitive()
}

File: net/JetBrains.Serialization/src/Dataholders/RdNInfo.cs
using Newtonsoft.Json;
using Org.BouncyCastle.Asn1;

namespace JetBrains.Serialization;

[JsonObject(MemberSerialization.OptIn)]
public class RdNInfo : IEncodableInfo
{
  [JsonProperty("Type")] private TextualInfo _type;
  [JsonProperty("Value")] private TextualInfo _value;

  [JsonConstructor]
  public RdNInfo(TextualInfo type, TextualInfo value)
  {
    _type = type;
    _value = value;
  }

  public Asn1Encodable ToPrimitive() =>
    new List<IEncodableInfo> { _type, _value }.ToPrimitiveDerSequence();
}